packages/web/src/components/OpenSourceDatasets/index.tsx

```diff
@@ -26,6 +35,40 @@ export default function OpenSourceDatasets() {
         );
     }, [dispatch]);
 
+    const openDatasetInApp = (
+        datasetName: string,
+        source: Source,
+        url?: Partial<FileExplorerURLComponents>
+    ) => {
+        dispatch(
```

Comment:
ooo interesting so this lets the state parse out the components provided by the URL while using the react router to migrate us between pages? Smart! I like that we don't have to have an explicit link that forces everything to reload